src/Cardano/Shell/Features/Logging.hs

```diff
@@ -57,33 +44,35 @@ testRotationParameters = RotationParameters
 -- the functions effects and constraining the user (programmer) of those function to use specific effects in them.
 -- https://github.com/input-output-hk/cardano-sl/blob/develop/util/src/Pos/Util/Log/LogSafe.hs
 data LoggingLayer = LoggingLayer
-    { iolLogDebug :: forall m. (MonadIO m) => Text -> m ()
-    , iolLogInfo  :: forall m. (MonadIO m) => Text -> m ()
-    , iolNonIo    :: forall m. (MonadThrow m) => m ()
-    }
-
-testLoggingLayer :: LoggingLayer
-testLoggingLayer = LoggingLayer
-    { iolLogDebug               = liftIO . putTextLn
-    , iolLogInfo                = liftIO . putTextLn
-    , iolNonIo                  = pure ()
+    { startTrace  :: forall m. (MonadIO m) => Trace m
+    , logDebug    :: forall m. (MonadIO m) => Trace m -> Text -> m ()
+    , logInfo     :: forall m. (MonadIO m) => Trace m -> Text -> m ()
+    , logNotice   :: forall m. (MonadIO m) => Trace m -> Text -> m ()
+    , logWarning  :: forall m. (MonadIO m) => Trace m -> Text -> m ()
+    , logError    :: forall m. (MonadIO m) => Trace m -> Text -> m ()
+    , appendName  :: forall m. (MonadIO m) => Text -> Trace m -> m (Trace m)
+    , mockNonIO   :: forall m. (MonadThrow m) => m ()
     }
 
 --------------------------------
 -- Feature
 --------------------------------
 
-type LoggingCardanoFeature = CardanoFeatureInit NoDependency RotationParameters LoggingLayer
+type LoggingCardanoFeature = CardanoFeatureInit NoDependency LoggingParameters LoggingLayer
 
 createLoggingFeature :: CardanoEnvironment -> CardanoConfiguration -> IO (LoggingLayer, CardanoFeature)
 createLoggingFeature cardanoEnvironment cardanoConfiguration = do
     -- we parse any additional configuration if there is any
     -- We don't know where the user wants to fetch the additional configuration from, it could be from
     -- the filesystem, so we give him the most flexible/powerful context, @IO@.
-    loggingConfiguration    <-  pure testRotationParameters
+    loggingConfiguration    <-  pure $ LoggingParameters "./config/logging.yaml"
```

Comment:
For example, why don't you parse the logging configuration here and pass that logging configuration to your _feature_?
That's the idea around it.
Also, what happens when you can't find this file?